api/v1/server/handlers/workflows/cancel.go

```diff
@@ -0,0 +1,103 @@
+package workflows
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/go-multierror"
+	"github.com/jackc/pgx/v5/pgtype"
+	"github.com/labstack/echo/v4"
+
+	"github.com/hatchet-dev/hatchet/api/v1/server/oas/gen"
+	"github.com/hatchet-dev/hatchet/api/v1/server/oas/transformers"
+	"github.com/hatchet-dev/hatchet/internal/msgqueue"
+	"github.com/hatchet-dev/hatchet/internal/repository/prisma/db"
+	"github.com/hatchet-dev/hatchet/internal/repository/prisma/dbsqlc"
+	"github.com/hatchet-dev/hatchet/internal/services/shared/tasktypes"
+)
+
+func (t *WorkflowService) WorkflowRunCancel(ctx echo.Context, request gen.WorkflowRunCancelRequestObject) (gen.WorkflowRunCancelResponseObject, error) {
+	tenant := ctx.Get("tenant").(*db.TenantModel)
+	runIds := request.Body.WorkflowRunIds
+
+	canceledStepRunsChannel := make(chan []*db.StepRunModel, len(runIds))
```

Comment:
While a channel can be used here, I think a `sync.WaitGroup` together with a result object like `sync.Map` would be more readable and easier to maintain.  With this implementation, you need to ensure that a message is sent over `canceledStepRunsChannel` on every branch of the goroutine. With a shared result object like `sync.Map` (or thread-safe slice implementation) together with `sync.WaitGroup`, you only need to write a single result to the result object.